### ISSUE TITLE
Updating configuration of a service where the status is currently 'critical' or 'warning' fails

### ISSUE THREAD

**Issue by paddie** [terraform-providers/terraform-provider-pagerduty#82]
It seems the Pagerduty API fails with a `400` when attempting to update the attributes of a `service` with an active incident on it resulting in its status being either `warning` or `critical`. This might just be a bug in the PagerDuty API though.

### Terraform Version
Terraform v0.11.6

### Affected Resource(s)
- `resource "pagerduty_service"`

### Terraform Configuration Files
```hcl
resource "pagerduty_service" "service" {
  name                    = "test"
  description             = "description"
  acknowledgement_timeout = "null"
  auto_resolve_timeout    = "null"
  escalation_policy       = "<ID>"
  alert_creation          = "create_alerts_and_incidents"

  incident_urgency_rule {
    type    = "constant"
    urgency = "high"
  }
}
```
### Expected Behavior
Successful update of the configuration of the service

### Actual Behavior
```
* module.business-systems.pagerduty_service.service: 1 error(s) occurred:

* pagerduty_service.service: PUT API call to https://api.pagerduty.com/services/PKS1O3C failed 400 Bad Request. Code: 2001, Errors: [Status must be in active or disabled.], Message: Invalid Input Provided
```

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply` - create service
2. change a single attribute, like the description of the service
3. re-apply while that service is in a state where it has an active incident on it.

**Comment by tmichelet** [comment]
+1, I'd be interested in a workaround

**Comment by pdecat** [comment]
I believe an HTTP PATCH operation would be more appropriate than a PUT to update an existing service's properties but I'm not sure it is supported as it is not documented: https://api-reference.pagerduty.com/#!/Services/put_services_id

**Comment by heimweh** [comment]
Hi @paddie, a [fix](https://github.com/terraform-providers/terraform-provider-pagerduty/pull/133) for this was merged and released in 1.3.1. 
I'll go ahead and close this for now but feel free to re-open at any time.
